src/test/lib/ApiTest.cpp

```diff
@@ -4227,6 +4227,116 @@ void QuicTest_QUIC_PARAM_CONN_STATISTICS_V2_PLAT(MsQuicRegistration& Registratio
     }
 }
 
+
+void QuicTest_QUIC_PARAM_CONN_ORIG_DEST_CID(MsQuicRegistration& Registration) {
+    //
+    // This is the unit test for checking to see if a server has the correct original dest CID.
+    //
+    TestScopeLogger LogScope0("QUIC_PARAM_CONN_ORIG_DEST_CID");
+    {
+        MsQuicConnection Connection(Registration);
+        TEST_QUIC_SUCCEEDED(Connection.GetInitStatus());
+        MsQuicAlpn Alpn("MsQuicTest");
+        MsQuicConfiguration ClientConfiguration(Registration, Alpn, ClientCertCredConfig);
+        TEST_QUIC_SUCCEEDED(
+        MsQuic->ConnectionStart(
+            Connection.Handle,
+            ClientConfiguration,
+            QUIC_ADDRESS_FAMILY_INET,
+            "localhost",
+            4433));
+        //
+        // 8 bytes is the expected minimum size of the CID.
+        //
+        uint32_t SizeOfBuffer = 8;
+        uint8_t Buffer[8]; 
+        TestScopeLogger LogScope1("GetParam test success case");
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_SUCCESS, 
+            Connection.GetParam(
+                QUIC_PARAM_CONN_ORIG_DEST_CID,
+                &SizeOfBuffer, 
+                Buffer
+            )
+        )
+    }
+    {
+        MsQuicConnection Connection(Registration);
+        TEST_QUIC_SUCCEEDED(Connection.GetInitStatus());
+        MsQuicAlpn Alpn("MsQuicTest");
+        MsQuicConfiguration ClientConfiguration(Registration, Alpn, ClientCertCredConfig);
+        TEST_QUIC_SUCCEEDED(
+        MsQuic->ConnectionStart(
+            Connection.Handle,
+            ClientConfiguration,
+            QUIC_ADDRESS_FAMILY_INET,
+            "localhost",
+            4433));
+        uint32_t SizeOfBuffer = 8;
+        TestScopeLogger LogScope1("GetParam null buffer check");
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER, 
+            Connection.GetParam(
+                QUIC_PARAM_CONN_ORIG_DEST_CID,
+                &SizeOfBuffer, 
+                nullptr
+            )
+        )
+    }
+    {
+        MsQuicConnection Connection(Registration);
+        TEST_QUIC_SUCCEEDED(Connection.GetInitStatus());
+        MsQuicAlpn Alpn("MsQuicTest");
+        MsQuicConfiguration ClientConfiguration(Registration, Alpn, ClientCertCredConfig);
+        TEST_QUIC_SUCCEEDED(
+        MsQuic->ConnectionStart(
+            Connection.Handle,
+            ClientConfiguration,
+            QUIC_ADDRESS_FAMILY_INET,
+            "localhost",
+            4433));
+        uint32_t SizeOfBuffer = 1;
+        TestScopeLogger LogScope1("GetParam buffer too small check");
+        uint8_t Buffer[1];
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_BUFFER_TOO_SMALL, 
+            Connection.GetParam(
+                QUIC_PARAM_CONN_ORIG_DEST_CID,
+                &SizeOfBuffer, 
+                Buffer
+            )
+        )
+    }
+    {
+        MsQuicConnection Connection(Registration);
```

Comment:
Looks like Configuration is prepared by caller function.
Try use it as 2nd argument of this func.

Reply:
Take the ClientConfiguration as the other test does. (e.g. line 4369)

Reply:
Will do!